### PR TITLE
Convert /web/javascript/reference/global_objects/remaining folder to Markdown (es)

### PR DESCRIPTION
**Pull request by Graywolf9**
This to continue the job already done in https://github.com/mdn/translated-content/pull/7618

### CODE REVIEW

File: files/es/web/javascript/reference/global_objects/generator/index.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+Este objeto no puede ser instanciado directamente. En su lugar, una instancia del objeto `Generator` puede ser devuelta por una [función generator](/es/docs/Web/JavaScript/Referencia/Sentencias/function*):
+
+    	function* gen() {
+    	  yield 1;
+    	  yield 2;
+    	  yield 3;
+    	}
+
+    	var g = gen(); // "Generator { }"
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/generator/index.md:29]:
````suggestion
```js
function* gen() {
    yield 1;
    yield 2;
    yield 3;
}

var g = gen(); // "Generator { }"
```
````

File: files/es/web/javascript/reference/global_objects/generator/index.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	    while(true)
+	        yield index++;
+	}
+
+	var gen = idMaker(); // "Generator { }"
+
+	console.log(gen.next().value); // 0
+	console.log(gen.next().value); // 1
+	console.log(gen.next().value); // 2
+	// ...
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/generator/index.md:68]:
```suggestion
	 function* idMaker() {
  var index = 0;
  while(true)
    yield index++;
}

var gen = idMaker(); // "Generator { }"

console.log(gen.next().value); // 0
console.log(gen.next().value); // 1
console.log(gen.next().value); // 2
// ...
```

File: files/es/web/javascript/reference/global_objects/generator/index.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+  `Generator.prototype.close()` {{non-standard_inline}}
+
+  - :
+    Cierra el generador. Sí el método next() es llamado por la expresión {{jsxref("StopIteration")}}, un error será lanzado. Esto se corresponde con el método return() del objeto generador de ES2015.
+
+- `Generator.prototype.send()` {{non-standard_inline}}
+  - : Utilizado para enviar un valor al generador. El valor es retornado por la expresion {{jsxref("Operators/yield", "yield")}}, y retorna el valor ofrecido por la siguiente expresion {{jsxref("Operators/yield", "yield")}}. `send(x)` Se corresponde a `next(x)` en el objeto generador de ECMAScript 2015.
+- **`Generator.`**`prototype.` **`throw()` **{{non-standard_inline}}
+  - : Lanza un error al generador. Esto se corresponde con el metodo throw() en el objeto genererador de ES2015.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/generator/index.md:89]:
```suggestion
- `Generator.prototype.next() `{{non-standard_inline}}
  - : Retorna el valor ofrecido por la expresión {{jsxref("Operators/yield", "yield")}}. Esto se corresponde con el método next() en el objeto generador de ES2015.
- `Generator.prototype.close()` {{non-standard_inline}}
  - : Cierra el generador. Sí el método next() es llamado por la expresión {{jsxref("StopIteration")}}, un error será lanzado. Esto se corresponde con el método return() del objeto generador de ES2015.
- `Generator.prototype.send()` {{non-standard_inline}}
  - : Utilizado para enviar un valor al generador. El valor es retornado por la expresion {{jsxref("Operators/yield", "yield")}}, y retorna el valor ofrecido por la siguiente expresion {{jsxref("Operators/yield", "yield")}}. `send(x)` Se corresponde a `next(x)` en el objeto generador de ECMAScript 2015.
- `Generator.prototype.throw()` {{non-standard_inline}}
  - : Lanza un error al generador. Esto se corresponde con el metodo throw() en el objeto genererador de ES2015.
```

File: files/es/web/javascript/reference/global_objects/generator/index.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	  yield a * 2;
+	}
+
+	var it = fibonacci();
+	console.log(it);          // "Generator {  }"
+	console.log(it.next());   // 1
+	console.log(it.send(10)); // 20
+	console.log(it.close());  // undefined
+	console.log(it.next());   // throws StopIteration (as the generator is now closed)
+
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/generator/index.md:105]:
```suggestion
function fibonacci() {
  var a = yield 1;
  yield a * 2;
}

var it = fibonacci();
console.log(it);          // "Generator {  }"
console.log(it.next());   // 1
console.log(it.send(10)); // 20
console.log(it.close());  // undefined
console.log(it.next());   // throws StopIteration (as the generator is now closed)
```

File: files/es/web/javascript/reference/global_objects/infinity/index.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+tags:
+  - JavaScript
+translation_of: Web/JavaScript/Reference/Global_Objects/Infinity
+original_slug: Web/JavaScript/Referencia/Objetos_globales/Infinity
+---
+{{jsSidebar("Objects")}}
+
+## Resumen
+
+La propiedad global **`Infinity `**es un valor numérico que representa el infinito.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/infinity/index.md:13]:
```suggestion
La propiedad global **`Infinity `** es un valor numérico que representa el infinito.
```

File: files/es/web/javascript/reference/global_objects/json/index.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+| Números            | Los ceros a la izquierda están prohibidos; un punto decimal debe ser seguido al menos por un dígito.                                                                                                                                                                                                                                                                                                                                                   |
+| Cadenas            | Solo un limitado conjunto de caracteres pueden ser de escape; ciertos caracteres de control estan prohibidos; los caracteres de separador de linea Unicode (U+2028) y el separador de parrafo (U+2029) son permitidos; las cadenas deben estar entre comillas dobles. Mira el siguiente ejemplo donde {{jsxref("JSON.parse")}} funciona bien y un{{jsxref("SyntaxError")}} es generado cuando se evalua el codigo como JavaScript:```js |
+
+var code = '"\u2028\u2029"'; JSON.parse(code); // works fine eval(code); // fails
+
+````|
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/json/index.md:33]:
```suggestion
| Cadenas            | Solo un limitado conjunto de caracteres pueden ser de escape; ciertos caracteres de control estan prohibidos; los caracteres de separador de linea Unicode (U+2028) y el separador de parrafo (U+2029) son permitidos; las cadenas deben estar entre comillas dobles. Mira el siguiente ejemplo donde {{jsxref("JSON.parse")}} funciona bien y un{{jsxref("SyntaxError")}} es generado cuando se evalua el codigo como JavaScript: <pre lang="js">var code = '"\u2028\u2029"';<br>JSON.parse(code); // works fine<br>eval(code); // fails</pre> |
```

File: files/es/web/javascript/reference/global_objects/json/index.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+    JSONObject = { }
+              or { Members }
+    Members = JSONString : JSON
+           or Members , JSONString : JSON
+
+    JSONArray = [ ]
+             or [ ArrayElements ]
+    ArrayElements = JSON
+                 or ArrayElements , JSON
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/json/index.md:83]:
```suggestion
```js
JSON = null
    or true or false
    or JSONNumber
    or JSONString
    or JSONObject
    or JSONArray

JSONNumber = - PositiveNumber
          or PositiveNumber
PositiveNumber = DecimalNumber
              or DecimalNumber . Digits
              or DecimalNumber . Digits ExponentPart
              or DecimalNumber ExponentPart
DecimalNumber = 0
              or OneToNine Digits
ExponentPart = e Exponent
            or E Exponent
Exponent = Digits
        or + Digits
        or - Digits
Digits = Digit
      or Digits Digit
Digit = 0 through 9
OneToNine = 1 through 9

JSONString = ""
          or " StringCharacters "
StringCharacters = StringCharacter
                or StringCharacters StringCharacter
StringCharacter = any character
                  except " or \ or U+0000 through U+001F
                or EscapeSequence
EscapeSequence = \" or \/ or \\ or \b or \f or \n or \r or \t
              or \u HexDigit HexDigit HexDigit HexDigit
HexDigit = 0 through 9
        or A through F
        or a through f

JSONObject = { }
          or { Members }
Members = JSONString : JSON
        or Members , JSONString : JSON

JSONArray = [ ]
          or [ ArrayElements ]
ArrayElements = JSON
              or ArrayElements , JSON
```
```

File: files/es/web/javascript/reference/global_objects/number/parseint/index.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+## Syntax
+
+    Number.parseInt(string,[ radix])
+
+### Parameters
+
+- `string`
+  - : The value to parse. If this argument is not a string, then it is converted to one using the [`ToString`](https://tc39.es/ecma262/#sec-tostring) abstract operation. Leading whitespace in this argument is ignored.
+- `radix`_ {{optional_inline}}_
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/number/parseint/index.md:25]:
```suggestion
- `radix` {{optional_inline}}
```

File: files/es/web/javascript/reference/global_objects/promise/all/index.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+- iterable
+  - : Un objeto iterable, por ejemplo un {{jsxref("Array")}}. Vea [iterable](/es/docs/Web/JavaScript/Guide/iterable).
+
+### Valor devuelto
+
+Una {{jsxref("Promise")}} que se cumplirá cuando todas las promesas del argumento iterable hayan sido cumplidas, o bien se rechazará cuando alguna de ellas se rechace.
+
+## Descripción
+
+**`Promise.all` **se cumple cuando todas las promesas del iterable dado se han cumplido, o es rechazada si alguna promesa no se cumple. {{jsxref("Promise.resolve")}}.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/promise/all/index.md:30]:
```suggestion
**`Promise.all`** se cumple cuando todas las promesas del iterable dado se han cumplido, o es rechazada si alguna promesa no se cumple. {{jsxref("Promise.resolve")}}.
```

File: files/es/web/javascript/reference/global_objects/promise/catch/index.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+        return originalThen.apply(this, arguments);
+    };
+    Promise.prototype.catch = function(){
+        console.log('> > > > > > called .catch on %o with arguments: %o', this, arguments);
+        return originalCatch.apply(this, arguments);
+    };
+
+})(this.Promise);
+
+
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/promise/catch/index.md:48]:
```suggestion
(function(Promise){
  var originalThen = Promise.prototype.then;
  var originalCatch = Promise.prototype.catch;

  Promise.prototype.then = function(){
    console.log('> > > > > > called .then on %o with arguments: %o', this, arguments);
    return originalThen.apply(this, arguments);
  };
  Promise.prototype.catch = function(){
    console.log('> > > > > > called .catch on %o with arguments: %o', this, arguments);
    return originalCatch.apply(this, arguments);
  };
})(this.Promise);
```

File: files/es/web/javascript/reference/global_objects/promise/catch/index.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+El método **catch()** retorna una `Promise` y solo se ejecuta en los casos en los que la promesa se marca como `Reject`. Se comporta igual que al llamar {{jsxref("Promise.then", "Promise.prototype.then(undefined, onRejected)")}} (de hecho, al llamar `obj.catch(onRejected)` internamente llama a `obj.then(undefined, onRejected)`).
+
+## Síntaxis
+
+    p.catch(onRejected);
+
+    p.catch(function(reason) {
+       // rejection
+    });
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/promise/catch/index.md:17]:
````suggestion
```js
p.catch(onRejected);

p.catch(function(reason) {
  // rejection
});
```
````

File: files/es/web/javascript/reference/global_objects/promise/catch/index.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+
+### Parámetros
+
+- onRejected
+  - : Una {{jsxref("Function")}} llamada cuando la `Promise` es rechazada. Esta función tiene un argumento:_ `reason`
+    _ : La razón del rechazo.La promesa devuelta por `catch()` es rechazada si `onRejected` lanza un error o retorna una `Promise` que a su vez se rechaza, de cualquier otra manera la `Promise` es resuelta.
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/promise/catch/index.md:23]:
```suggestion
- onRejected
  - : Una {{jsxref("Function")}} llamada cuando la `Promise` es rechazada. Esta función tiene un argumento:
    - `reason`
      - : La razón del rechazo.
    La promesa devuelta por `catch()` es rechazada si `onRejected` lanza un error o retorna una `Promise` que a su vez se rechaza, de cualquier otra manera la `Promise` es resuelta.
```

File: files/es/web/javascript/reference/global_objects/promise/then/index.md [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+
+## Sintaxis
+
+    p.then(alCumplir[, enRechazo]);
+
+    p.then(function(value) {
+       // cumplimiento
+      }, function(reason) {
+      // rechazo
+    });
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/promise/then/index.md:27]:
````suggestion
```js
p.then(alCumplir[, enRechazo]);

p.then(function(value) {
  // cumplimiento
}, function(reason) {
  // rechazo
});
```
````

File: files/es/web/javascript/reference/global_objects/promise/then/index.md [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+
+Retorna un `Promise` el cual es determinado por las funciones input:
+
+- Si alCumplir o `enRechazo` arroja un error, o retorna un [`Promise`](/es/docs/Web/JavaScript/Referencia/Objetos_globales/Promise) rechazado, `then` retorna un `Promise` rechazado.
+- Si alCumplir o `enRechazo` retorna un `Promise` que resuelve, o retorna cualquier otro valor, `then` retorna un `Promise `resuelto.
+
+- alCumplir {{optional_inline}}
+  - : Una [Función](es/docs/Web/JavaScript/Referencia/Objetos_globales/Function) es llamada si la `Promesa` se cumple. Esta función tiene un argumento, el `valor de` cumplimiento. Si no es una función, se reemplaza internamente con una función de "Identidad" (devuelve el argumento recibido).
+- `enRechazo` {{optional_inline}}
+  - : Una [Función](es/docs/Web/JavaScript/Referencia/Objetos_globales/Function) es llamada si la `Promesa `es rechazada. Esta función tiene un argumento, la `razón` de rechazo. Si no es una función, se reemplaza internamente con una función "Lanzador" (lanza un error que recibió como argumento).
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/promise/then/index.md:39]:
```suggestion
* Si `alCumplir` o `enRechazo` arroja un error, o retorna un [`Promise`](/es/docs/Web/JavaScript/Referencia/Objetos_globales/Promise) rechazado, `then` retorna un `Promise` rechazado.
* Si `alCumplir` o `enRechazo` retorna un `Promise` que resuelve, o retorna cualquier otro valor, `then` retorna un `Promise `resuelto.

- `alCumplir` {{optional_inline}}
  - : Una [Función](es/docs/Web/JavaScript/Referencia/Objetos_globales/Function) es llamada si la `Promesa` se cumple. Esta función tiene un argumento, el `valor de` cumplimiento. Si no es una función, se reemplaza internamente con una función de "Identidad" (devuelve el argumento recibido).
- `enRechazo` {{optional_inline}}
  - : Una [Función](es/docs/Web/JavaScript/Referencia/Objetos_globales/Function) es llamada si la `Promesa `es rechazada. Esta función tiene un argumento, la `razón` de rechazo. Si no es una función, se reemplaza internamente con una función "Lanzador" (lanza un error que recibió como argumento).
```

File: files/es/web/javascript/reference/global_objects/regexp/compile/index.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+
+    regexObj.compile(patrón, flags)
+
+### Parámetros
+
+- `patrón`
+  - : El texto de la expresión regular.
+- `flags`
+  - : Si es especificado, las flags pueden tener cualquier combinación de los siguientes valores:_ `g`
+    _ : búsqueda global (global match)
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/regexp/compile/index.md:30]:
```suggestion
  - : Si es especificado, las flags pueden tener cualquier combinación de los siguientes valores:
    - `g`
      - : búsqueda global (global match)
```

File: files/es/web/javascript/reference/global_objects/escape/index.md [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+tags:
+  - JavaScript
+  - Obsoleto
+  - escape
+translation_of: Web/JavaScript/Reference/Global_Objects/escape
+original_slug: Web/JavaScript/Referencia/Objetos_globales/escape
+---
+> **Advertencia:** {{Deprecated_header}} Advertencia `escape()` no esta estrictamente en desuso("eliminada por los estándares Web"), esta definida en [Anexo B](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-additional-ecmascript-features-for-web-browsers) El estándar ECMA-262 , cuya introducción establece:> … Todos las características especificas y comportamientos del lenguaje en este anexo tienen mas de una caracterísitca indeseable y en ausencia del legado sera eliminada de esta especificación. …
+>
+> > … Los programadores no deben usar o suponer la existencia de estas características y/o comportamientos al escribir nuevo código ECMAScript …
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/escape/index.md:13]:
```suggestion
{{Deprecated_header}}

> **Advertencia:**  `escape()` no esta estrictamente en desuso("eliminada por los estándares Web"), esta definida en [Anexo B](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-additional-ecmascript-features-for-web-browsers) El estándar ECMA-262 , cuya introducción establece:
> > … Todos las características especificas y comportamientos del lenguaje en este anexo tienen mas de una caracterísitca indeseable y en ausencia del legado sera eliminada de esta especificación. …
> > 
> > … Los programadores no deben usar o suponer la existencia de estas características y/o comportamientos al escribir nuevo código ECMAScript …
```

File: files/es/web/javascript/reference/global_objects/undefined/index.md [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+>
+> // registra "foo string"
+> (function(){ var undefined = 'foo'; console.log(undefined, typeof undefined); })()
+>
+> // registra "foo string"
+> (function(undefined){ console.log(undefined, typeof undefined); })('foo');
+>
+> ```
+>
+> ```
```
Comment (Graywolf9) [inline files/es/web/javascript/reference/global_objects/undefined/index.md:42]:
```suggestion
> **Advertencia:** Aunque es posible usarlo como un {{Glossary("Identifier", "identificador")}} (nombre de variable) en cualquier otro ámbito que no sea el ámbito global (porque `undefined `no es una {{jsxref("Reserved_Words", "palabra reservada")}}), hacerlo es una mala idea que provocará que tú código sea difícil de mantener y depurar.
> ```js
> //NO HAGAS ESTO
>
> // registra "foo string"
> (function(){ var undefined = 'foo'; console.log(undefined, typeof undefined); })()
>
> // registra "foo string"
> (function(undefined){ console.log(undefined, typeof undefined); })('foo');
> ```
```

**Comment by OnkarRuikar** [comment]
The PR [redirects check is complaining](https://github.com/mdn/translated-content/actions/runs/3114446745/jobs/5050295335) about two entries in the `_redirects.txt` file:
```log
removing orphaned redirect (from exists): /es/docs/Web/JavaScript/Reference/Global_Objects/WeakMap/clear	/es/docs/conflicting/Web/JavaScript/Reference/Deprecated_and_obsolete_features
removing orphaned redirect (from exists): /es/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly	/es/docs/WebAssembly/JavaScript_interface
info: _redirects.txt for es is causing issues: Error: Invalid redirect for /es/docs/Web/JavaScript/Reference/Global_Objects/WeakMap/clear -> /es/docs/conflicting/Web/JavaScript/Reference/Deprecated_and_obsolete_features or /es/docs/Web/JavaScript/Reference/Global_Objects/null -> /es/docs/Web/JavaScript/Reference/Operators/null
```

The two entries have become invalid because you've added these files back to the repo in the PR. 
Removing these entries resolves the issue.

Save following content in `fixRedirects.patch` file and run command `git apply fixRedirects.patch` in your local.
```patch
diff --git a/files/es/_redirects.txt b/files/es/_redirects.txt
index d9b74ceb7b..74d5b41219 100644
--- a/files/es/_redirects.txt
+++ b/files/es/_redirects.txt
@@ -2217,8 +2217,6 @@
 /es/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler/getOwnPropertyDescriptor	/es/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/getOwnPropertyDescriptor
 /es/docs/Web/JavaScript/Reference/Global_Objects/Proxy/handler/set	/es/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/set
 /es/docs/Web/JavaScript/Reference/Global_Objects/String/toSource	/es/docs/conflicting/Web/JavaScript/Reference/Global_Objects/String/toString
-/es/docs/Web/JavaScript/Reference/Global_Objects/WeakMap/clear	/es/docs/conflicting/Web/JavaScript/Reference/Deprecated_and_obsolete_features
-/es/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly	/es/docs/WebAssembly/JavaScript_interface
 /es/docs/Web/JavaScript/Reference/Global_Objects/null	/es/docs/Web/JavaScript/Reference/Operators/null
 /es/docs/Web/JavaScript/Reference/Statements/import.meta	/es/docs/Web/JavaScript/Reference/Operators/import.meta
 /es/docs/Web/JavaScript/Referencia	/es/docs/Web/JavaScript/Reference
```
Or if you are comfortable, manually delete line 2220 and 2221 from the `/files/es/_redirects.txt` file.

**Comment by Graywolf9** [comment]
Thank you so much for your time and help! @OnkarRuikar I couldn't apply the patch, but deleting the lines works!

**Pull request merged**